hydra/utils.py

```diff
@@ -43,6 +43,11 @@ def get_static_method(full_method_name):
 
 def instantiate(config, *args, **kwargs):
     assert config is not None, "Input config is None"
+    assert isinstance(
+        config.params, (DictConfig, ListConfig)
```

Comment:
@omry The reason why I'm keeping `ListConfig` is that [OmegaConf handles that](https://github.com/omry/omegaconf/blob/6c7c0f044281cdc1cecfebdce385466b7f48559b/omegaconf/config.py#L396) by raising 

> Merging DictConfig with ListConfig is not supported

The exception says "not supported" and doesn't throw a semantic error. If later on `OmegaConf` decides to add that support, we would still be limiting that in the `Hydra` layer even though it would work.